### PR TITLE
Made the repository RESUE-compliant

### PR DESCRIPTION
**Pull request by silverhook**
see <https://reuse.software> for details

Connected with https://github.com/fish-shell/fish-shell/pull/9326

### CODE REVIEW

**Comment by faho** [comment]
Personally I would again prefer a big file for the tests - here those aren't typically installed, the way to get littlecheck in your project is to just copy littlecheck.py. So keep the header in that and add a file that marks *all files* with CC0-1.0.

Also, because this technically involves relicensing (even tho the previous license would allow this one), I would like to have agreement from @ridiculousfish, @mqudsi and @krobelus that that's okay. It's fine by me, and I'm uncomfortable with the current public domain declaration anyway because that's awkward under german law.

(the only other contributor only removed a line so they don't have copyright)

File: LICENSES/CC0-1.0.txt [hunk 1]
```diff
@@ -0,0 +1,121 @@
+Creative Commons Legal Code
+
+CC0 1.0 Universal
```
Comment (krobelus) [inline LICENSES/CC0-1.0.txt:3]: what's the problem with the public domain license in the existing LICENSE file?
I know that public domain is contentious (why?) but it seems like a good license for this sort of thing.

Comment (faho) [reply LICENSES/CC0-1.0.txt:3]: Public Domain is legally problematic because it's not a license, it's you giving up your rights.

That's not allowed in a bunch of jurisdictions (like the EU), which is why CC0 exists.

**Comment by ridiculousfish** [comment]
I'm in favor of adding a license declaration to littlecheck.py, and/or a centralized license file, but modifying all the test files seems unnecessary to me.

Our README says "littlecheck is released into the public domain. You may use it for any purpose without attribution, or under the [CC0](https://creativecommons.org/publicdomain/zero/1.0/) license if public domain is not available."

creativecomments [recommends the following language](https://wiki.creativecommons.org/wiki/CC0_FAQ#May_I_apply_CC0_to_computer_software.3F_If_so.2C_is_there_a_recommended_implementation.3F) to apply CC0:

> To the extent possible under law, the author(s) have dedicated all copyright and related and neighboring rights to this software to the public domain worldwide. This software is distributed without any warranty.

this seems like a fine wording to me, since it places it in public domain in jurisdictions where that's allowed. So I'm in favor of this change.

So basically:

1. Add the license info to littlecheck.py
2. Add a centralized license file, perhaps a COPYING.txt
3. Change the verbiage in the README to emphasize CC0 and mention public domain, perhaps "littlecheck is licensed under CC0, which places it in the public domain to the extent possible by law."
